lib/internal/modules/cjs/loader.js

```diff
@@ -797,6 +800,7 @@ Module._load = function(request, parent, isMain) {
       throw new ERR_UNKNOWN_BUILTIN_MODULE(request);
     }
 
+    onModuleLoad.publish(module);
```

Comment:
Just like in [http_server.js](https://github.com/nodejs/node/blob/main/lib/_http_server.js#L1003), maybe we should use:
```js
if (channel.hasSubscribers) {
  channel.publish()
}
```

Reply:
The `hasSubscribers` check is only relevant if you're building input objects. If you just pass stuff in as-is it will be a no-op when there's no subscribers. Do we want to match anyway though just for consistency?

Reply:
Oh right my bad, you're right!